--- mesa-17.1.1/src/gallium/drivers/swr/SConscript.orig	2017-06-08 12:05:53.000000000 +0200
+++ mesa-17.1.1/src/gallium/drivers/swr/SConscript	2017-06-08 12:05:09.000000000 +0200
@@ -181,7 +181,7 @@
 envavx = env.Clone()
 
 envavx.Append(CPPDEFINES = ['KNOB_ARCH=KNOB_ARCH_AVX'])
-if env['platform'] == 'windows':
+if env['platform'] == 'windows' and env['CC'] != 'gcc':
     envavx.Append(CCFLAGS = ['/arch:AVX'])
 else:
     envavx.Append(CCFLAGS = ['-mavx'])
@@ -197,7 +197,7 @@
 envavx2 = env.Clone()
 
 envavx2.Append(CPPDEFINES = ['KNOB_ARCH=KNOB_ARCH_AVX2'])
-if env['platform'] == 'windows':
+if env['platform'] == 'windows' and env['CC'] != 'gcc':
     envavx2.Append(CCFLAGS = ['/arch:AVX2'])
 else:
     envavx2.Append(CCFLAGS = ['-mavx2', '-mfma', '-mbmi2', '-mf16c'])
